src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2431,7 +2431,8 @@ void primitive_inst::update_weights() {
             reorder_kernel_params->get_output_layout().clone_with_other_shape(original_layout.get_partial_shape());
         _impl_params->weights_layout = optional_layout(expected_layout);
 
-        if (_reordered_weights_cache.has(expected_layout)) {
+        if (_reordered_weights_cache.has(expected_layout) &&
+            !(expected_layout.format == cldnn::format::custom && expected_layout.bytes_count() > _reordered_weights_cache.get(expected_layout)->size())) {
```

Comment:
Should format_traits be compared instead of memory size when the format is custom?